### PR TITLE
Support http query parameter based hash

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Fixes: #20794 

depend on https://github.com/istio/api/pull/1272

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
Need to bump the API

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -972,6 +972,14 @@ func consistentHashToHashPolicy(consistentHash *networking.LoadBalancerSettings_
 				},
 			},
 		}
+	case *networking.LoadBalancerSettings_ConsistentHashLB_HttpQueryParameterName:
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/route/route.go]: Can you please add a test case in `TestBuildHttpRoutes` to cover this new setting?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: Will do

**Comment by hzxuzhonghu** [comment]
@howardjohn Any idea why `TestIOPController_SwitchProfile`  failed?

**Comment by jackkleeman** [comment]
Should be ok now if we update api dependency to https://github.com/istio/api/pull/1283

File: pilot/pkg/networking/core/v1alpha3/route/route.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: done

**Pull request merged**